src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/UploadGeneralDocumentsAboutToSubmitHandler.java

```diff
@@ -0,0 +1,70 @@
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.finrem.caseorchestration.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.helper.UploadedGeneralDocumentHelper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.CaseType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.GeneralUploadedDocument;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.GeneralUploadedDocumentData;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Comparator.comparing;
+import static java.util.Comparator.nullsLast;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.GENERAL_UPLOADED_DOCUMENTS;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class UploadGeneralDocumentsAboutToSubmitHandler implements CallbackHandler {
+
+    private final ObjectMapper objectMapper;
+    private final UploadedGeneralDocumentHelper uploadedGeneralDocumentHelper;
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType eventType) {
+        return CallbackType.ABOUT_TO_SUBMIT.equals(callbackType)
+            && EventType.UPLOAD_GENERAL_DOCUMENT.equals(eventType);
+    }
+
+    @Override
+    public AboutToStartOrSubmitCallbackResponse handle(CallbackRequest callbackRequest, String userAuthorisation) {
+
+        Map<String, Object> caseData = uploadedGeneralDocumentHelper.addUploadDateToNewDocuments(
+            callbackRequest.getCaseDetails().getData(),
+            callbackRequest.getCaseDetailsBefore().getData(),
+            GENERAL_UPLOADED_DOCUMENTS);
+
+        List<GeneralUploadedDocumentData> uploadedDocuments = getGeneralDocumentCollection(caseData);
+
+        uploadedDocuments.sort(comparing(GeneralUploadedDocumentData::getGeneralUploadedDocument,
+            comparing(GeneralUploadedDocument::getGeneralDocumentUploadDateTime, nullsLast(Comparator.reverseOrder()))));
+
+        caseData.put(GENERAL_UPLOADED_DOCUMENTS, uploadedDocuments);
+        return AboutToStartOrSubmitCallbackResponse
+            .builder()
+            .data(caseData)
+            .build();
+    }
+
+    private List<GeneralUploadedDocumentData> getGeneralDocumentCollection(Map<String, Object> caseData) {
+        objectMapper.registerModule(new JavaTimeModule());
+
+        List<GeneralUploadedDocumentData> generalDocuments = objectMapper.convertValue(
+            caseData.get(GENERAL_UPLOADED_DOCUMENTS), new TypeReference<>() {});
+
+        return Optional.ofNullable(generalDocuments).orElse(new ArrayList<>());
```

Comment:
Check if you can use emptyArray if you want this to be a immutable list returned